### ISSUE TITLE
Separate namespace for key functions

### ISSUE THREAD

**Issue by iptq** [marcusklaas/sortedvec#26]
I can't declare two `sortedvec`s in the same module using the same `keyfn` since the macro just drops them into the same module. Compartmentalizing them into thier own namespaces would allow this to work(or even just having it be a function associated with the struct, i may play around with this ina bit)

**Comment by marcusklaas** [comment]
I am afraid I don't quite understand the issue. I've tried to reproduce it as follows, but that seems to work without problem:
```rust
use sortedvec::sortedvec;

sortedvec! {
    struct SortedVec {
        fn derive_key(x: &u32) -> u32 { *x }
    }
}

sortedvec! {
    struct SortedVecTwo {
        fn derive_key(x: &u32) -> u32 { *x }
    }
}

fn main() {
    let unsorted = vec![3, 5, 0, 10, 7, 1];
    let sorted = SortedVecTwo::from(unsorted.clone());

    // binary search (fast!)
    let sorted_contains_six: Option<_> = sorted.find(&6);
    assert!(sorted_contains_six.is_none());
}
```
It's admittedly kind of awkward that this requires two macro invocations, so maybe that's tripping you up?

Could you post a (simplified) example which fails to compile?

**Comment by marcusklaas** [comment]
Ah I see this was indeed a problem in the past and should have been fixed on 0.5.0. I will publish this version shortly.

**Comment by marcusklaas** [comment]
The new version [has been published](https://crates.io/crates/sortedvec/0.5) and I'm optimistically closing this issue. Please reopen if the problem persists!
